baseapp/baseapp.go

```diff
@@ -829,10 +829,12 @@ func (app *BaseApp) runMsgs(ctx sdk.Context, msgs []sdk.Msg, mode runTxMode) (*s
 			err          error
 		)
 
+		msgCtx, msgMsCache := app.cacheTxContext(ctx, []byte{})
```

Comment:
The messages should not be committed in general as it's not being returned in the Tx result, but this branch allows us to make sure that the events in the CacheKv store only reflects the access ops for this message, so we can compare it per message, since that's how access ops are defined